src/invert/laplace/impls/petsc3damg/petsc3damg.cxx

```diff
@@ -275,12 +275,23 @@ Field3D LaplacePetsc3dAmg::solve(const Field3D &b_in, const Field3D &x0) {
       }
     }
   }
-  for (int b = 1; b < localmesh->xstart; b++) {
-    BOUT_FOR(i, indexer->getRegionInnerX()) {
-      solution[i.xm(b)] = solution[i];
+
+  // Set inner boundary cells by extrapolating
+  // from single boundary cell which is set by the solver
+  // Note: RegionInnerX is the set of points just outside the domain
+  //       (in the first boundary cell) so one boundary cell is already set
+  BOUT_FOR(i, indexer->getRegionInnerX()) {
+    for (int b = 1; b < localmesh->xstart; b++) {
+      solution[i.xm(b)] = 2.*solution[i.xm(b-1)] - solution[i.xm(b-2)];
```

Comment:
To be consistent with the extrapolation in the boundary conditions, e.g. here (after rearranging to replace `delta * val` with values of `f` from the previous line)
https://github.com/boutproject/BOUT-dev/blob/ca49afe984ee93203ce0709d30c9219bbde17a68/src/mesh/boundary_standard.cxx#L2138-L2143
I think we need higher order extrapolation:
```suggestion
      solution[i.xm(b)] = 3.*solution[i.xm(b-1)] - 3.*solution[i.xm(b-2)] + solution[i.xm(b-3)];
```